lib/main.dart

```diff
@@ -1,7 +1,10 @@
 import 'package:flutter/material.dart';
 
 void main() {
-  runApp(BooksApp());
+  runApp(RootRestorationScope(
```

Comment:
We need to manually create a root scope (usually this is created by MaterialApp, but we have state to restore in BooksApp which is an ancestor of MaterialApp and doesn't have access to the root scope created by it).